### PR TITLE
location-retrieval: Alignment of errors with Commonalities

### LINKED ISSUE #219: Align Location Retrieval with guidelines for error scenarios
**Problem description**
Current version still does not follow the recent guidelines.

**Expected behavior**
Align codes and examples as indicated in:

https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#6-error-responses
https://github.com/camaraproject/Commonalities/blob/main/artifacts/CAMARA_common.yaml

**Additional context**
Issue #214 tackled same topic for Location Verification

### PR DESCRIPTION
**Pull request by bigludo7**
#### What type of PR is this?

Add one of the following kinds:

* enhancement/feature

#### What this PR does / why we need it:

Alignement of errors according to Commonalities guideline.

#### Which issue(s) this PR fixes:

Fixes #219 

#### Changelog input

```
 release-note
- Aligned error code list & description with Commonalities

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

File: code/API_definitions/location-retrieval.yaml [hunk 1]
```diff
@@ -348,55 +348,83 @@ components:
...
-            InvalidArgument:
+            GENERIC_400_INVALID_ARGUMENT:
+              summary: Generic Invalid Argument
+              description: Invalid Argument. Generic Syntax Exception
               value:
                 status: 400
                 code: INVALID_ARGUMENT
-                message: "Invalid argument"
-            MaxAgeIssue:
+                message: Client specified an invalid argument, request body or query param.
```
Comment (alpaycetin74) [inline code/API_definitions/location-retrieval.yaml:369]: In theory this also applies to request headers. Maybe we can simplify the sentence, keep it more generic ?

File: code/API_definitions/location-retrieval.yaml [hunk 2]
```diff
@@ -348,55 +348,83 @@ components:
...
-            $ref: '#/components/schemas/ErrorInfo'
-          example:
-            status: 403
-            code: PERMISSION_DENIED
-            message: "Operation not allowed: ..."
+            $ref: "#/components/schemas/ErrorInfo"
+          examples:
+            GENERIC_403_PERMISSION_DENIED:
+              summary: Generic Permission Denied
+              description: Permission denied. OAuth2 token access does not have the required scope or when the user fails operational security
```
Comment (alpaycetin74) [inline code/API_definitions/location-retrieval.yaml:415]: Maybe reword as "OAuth2 token does not have access to the required scope, or the user fails operational security" ?

File: code/API_definitions/location-retrieval.yaml (hunk 1 above)
Comment (bigludo7) [reply code/API_definitions/location-retrieval.yaml:369]: Hello @alpaycetin74 
I used the the template from commonalities [here](https://github.com/camaraproject/Commonalities/blob/main/artifacts/CAMARA_common.yaml) - line 146.... so probably we cannot change this.

Comment (jlurien) [reply code/API_definitions/location-retrieval.yaml:369]: Also, the value for message in examples are not normative, only the code. Implementation may return any other, even in different language

File: code/API_definitions/location-retrieval.yaml [hunk 3]
```diff
@@ -407,16 +435,21 @@ components:
...
             $ref: '#/components/schemas/ErrorInfo'
           examples:
-            notFound:
-              value:
-                status: 404
-                code: NOT_FOUND
-                message: 'The specified resource is not found'
-            deviceNotFound:
+            LOCATION_RETRIEVAL_404_UNABLE_TO_LOCATE_DEVICE:
+              summary: The location server is not able to locate the mobile
```
Comment (jlurien) [inline code/API_definitions/location-retrieval.yaml:439]: Maybe better "device" instead of "mobile"

**Review by jlurien — APPROVED** [review]
Looks good, apart from reference to mobile

File: code/API_definitions/location-retrieval.yaml (hunk 3 above)
Comment (bigludo7) [reply code/API_definitions/location-retrieval.yaml]: Yes - fixed.

**Comment by bigludo7** [comment]
Fixed last @jlurien comment.
José, @alpaycetin74 we can probably move forward with this one.

**Comment by alpaycetin74** [comment]
> Fixed last @jlurien comment. José, @alpaycetin74 we can probably move forward with this one.

I agree, thank you.

**Comment by bigludo7** [comment]
@jlurien May I ask you to take a look on this one. I guess we need to move quick on this one for the meta-release. Thanks

**Review by jlurien — DISMISSED** [review]
We may need to add support for 422 UNIDENTIFIABLE_DEVICE both this one and verification, but we can ammend that once Commonalities issues is addressed

**Comment by bigludo7** [comment]
> We may need to add support for 422 UNIDENTIFIABLE_DEVICE both this one and verification, but we can ammend that once Commonalities issues is addressed

Added - If you think it's too early I can remove but Rafal is working on the PR.

**Review by jlurien — APPROVED** [review]
